### PR TITLE
Add cassandra state store

### LINKED ISSUE #45: Cassandra state store
## Describe the proposal

Implement Cassandra as a state store and transactional state store.

### PR DESCRIPTION
**Pull request by yaron2**
closes #45

### CODE REVIEW

File: state/cassandra/cassandra.go [hunk 1]
```diff
@@ -0,0 +1,304 @@
...
+
+// Set saves state into cassandra
+func (c *Cassandra) Set(req *state.SetRequest) error {
+	var bt []byte
+	b, ok := req.Value.([]byte)
+	if ok {
+		bt = b
+	} else {
+		bt, _ = jsoniter.ConfigFastest.Marshal(req.Value)
+	}
```
Comment (youngbupark) [inline state/cassandra/cassandra.go]: I think we used this logic multiple times. Wouldn't it be good to have utils package in component-contrib ?

Comment (yaron2) [reply state/cassandra/cassandra.go]: Yes, but lets save that for a separate refactor

File: state/cassandra/cassandra.go [hunk 2]
```diff
@@ -0,0 +1,304 @@
...
+	return nil
+}
+
+// Get retrieves state from cassandra with a key
+func (c *Cassandra) Get(req *state.GetRequest) (*state.GetResponse, error) {
+	query := fmt.Sprintf("SELECT value FROM %s WHERE key='%s'", c.table, req.Key)
+
+	session := c.session
+
+	if req.Options.Consistency == state.Strong {
```
Comment (youngbupark) [inline state/cassandra/cassandra.go]: Isn't there any chance that user passes unsupported consistency ? e.g. typo. evntual

File: state/cassandra/cassandra.go [hunk 3]
```diff
@@ -0,0 +1,304 @@
...
+		sess, err := c.createSession(gocql.Any)
+		if err != nil {
+			return err
+		}
+		defer sess.Close()
+		session = sess
+	}
+
+	err := session.Query(query, req.Key, bt).Exec()
+	return err
```
Comment (youngbupark) [inline state/cassandra/cassandra.go]: I would just return session.Query(x) without the assignment.
```
return session.Query(query, req.Key, bt).Exec()
```

File: state/cassandra/cassandra.go (hunk 2 above)
Comment (yaron2) [reply state/cassandra/cassandra.go]: Yes. And then the default consistency level is used

File: state/cassandra/cassandra.go (hunk 3 above)
Comment (yaron2) [reply state/cassandra/cassandra.go]: Alright...

Comment (yaron2) [reply state/cassandra/cassandra.go]: Fixed with commit

**Pull request merged**